engine/access/state_stream/backend/backend_executiondata2_test.go

```diff
@@ -0,0 +1,1065 @@
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+
+	"github.com/onflow/flow-go/access"
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/engine/access/subscription"
+	"github.com/onflow/flow-go/engine/access/subscription/tracker"
+	trackermock "github.com/onflow/flow-go/engine/access/subscription/tracker/mock"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module/executiondatasync/execution_data"
+	"github.com/onflow/flow-go/module/executiondatasync/optimistic_sync"
+	"github.com/onflow/flow-go/module/executiondatasync/optimistic_sync/execution_result"
+	osyncmock "github.com/onflow/flow-go/module/executiondatasync/optimistic_sync/mock"
+	"github.com/onflow/flow-go/module/irrecoverable"
+	protocolmock "github.com/onflow/flow-go/state/protocol/mock"
+	"github.com/onflow/flow-go/storage"
+	storagemock "github.com/onflow/flow-go/storage/mock"
+	"github.com/onflow/flow-go/utils/unittest"
+	"github.com/onflow/flow-go/utils/unittest/fixtures"
+)
+
+type BackendExecutionDataSuite2 struct {
+	suite.Suite
+
+	fixtureGenerator *fixtures.GeneratorSuite
+
+	log      zerolog.Logger
+	state    *protocolmock.State
+	params   *protocolmock.Params
+	snapshot *protocolmock.Snapshot
+	headers  *storagemock.Headers
+	receipts *storagemock.ExecutionReceipts
+
+	// execution data stuff
+	executionDataBroadcaster *engine.Broadcaster
+	subscriptionFactory      *subscription.SubscriptionHandler
+	executionDataTracker     *trackermock.ExecutionDataTracker
+	executionDataTrackerReal tracker.ExecutionDataTracker
+
+	// optimistic sync stuff
+	executionResultProvider     *osyncmock.ExecutionResultInfoProvider
+	executionResultProviderReal optimistic_sync.ExecutionResultInfoProvider
+	executionStateCache         *osyncmock.ExecutionStateCache
+	executionDataSnapshot       *osyncmock.Snapshot
+	criteria                    optimistic_sync.Criteria
+
+	// data for the tests
+	sporkRootBlock *flow.Block
+
+	blocks                 []*flow.Block
+	blocksHeightToBlockMap map[uint64]*flow.Block
+	blocksIDToBlockMap     map[flow.Identifier]*flow.Block
+
+	executionDataList         []*execution_data.BlockExecutionData
+	blockIDToExecutionDataMap map[flow.Identifier]*execution_data.BlockExecutionData
+
+	executionResults []*flow.ExecutionResult
+
+	// execution node configuration
+	fixedExecutionNodes       flow.IdentityList
+	preferredExecutionNodeIDs flow.IdentifierList
+}
+
+// TestBackendExecutionDataSuite2 runs the BackendExecutionDataSuite2 test suite.
+func TestBackendExecutionDataSuite2(t *testing.T) {
+	suite.Run(t, new(BackendExecutionDataSuite2))
+}
+
+// SetupTest initializes the test suite with necessary mocks and fixtures.
+func (s *BackendExecutionDataSuite2) SetupTest() {
+	s.log = unittest.Logger()
+
+	s.fixtureGenerator = fixtures.NewGeneratorSuite(
+		fixtures.WithChainID(flow.Testnet),
+		fixtures.WithSeed(42),
```

Comment:
let's avoid using a seed. this fixes the randomness so every run will get identical data. we want the data to be random in general.
```suggestion
```